src/logic/safe/store/actions/processTransaction.ts

```diff
@@ -160,18 +160,18 @@ export const processTransaction =
 
           if (isExecution) {
             dispatch(updateTransactionStatus({ safeTxHash: tx.safeTxHash, status: LocalTransactionStatus.PENDING }))
+            aboutToExecuteTx.setNonce(txArgs.nonce)
           }
 
-          try {
-            await saveTxToHistory({ ...txArgs })
-
-            // store the pending transaction's nonce
-            isExecution && aboutToExecuteTx.setNonce(txArgs.nonce)
-
-            dispatch(fetchTransactions(chainId, safeAddress))
-          } catch (e) {
-            logError(Errors._804, e.message)
+          if (!isExecution) {
+            try {
+              await saveTxToHistory({ ...txArgs })
+            } catch (e) {
+              logError(Errors._804, e.message)
+            }
           }
+
+          dispatch(fetchTransactions(chainId, safeAddress))
```

Comment:
I don't think you need to fetch transactions immediately if it was an execution because it will take time for the indexer to pick up the transaction. So the minimum would be the block time (10s).

Reply:
@katspaugh and I are in the process of doing a huge refactor of this and it is already removed there, but I'll add it here for good measure.